### PR TITLE
Fix Serial Monitor for VS Code 1.43.0

### PR DESCRIPTION
**Pull request by vandyliu**
[AB#34114](https://microsoftgarage.visualstudio.com/002806e2-ebaa-4672-9d2e-5fe5d29154ef/_workitems/edit/34114)

PLEASE TEST!!! Dependencies were a little flaky before so I want to make sure they're good now.

# Description:
- Uses "usb-native" npm package instead of "vendor/native-usb-native", which follows what the vscode-arduino team is doing in their PR (https://github.com/microsoft/vscode-arduino/pull/982/)
- Removed "vendor/native-usb-native"
- Updated minimum version of VS Code to 1.43.0 to accommodate using "usb-native" package. In VS Code 1.43.0, they changed the electron version so the serial monitor fix we are using here does not work in older versions of VS Code (I tried with 1.42.0 and it didn't work) 
  - VS Code auto updates to latest version

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)

# Limitations:

Please describe limitations of this PR

# Testing:

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- Tested with a micro:bit
- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] My code has been formatted with `npm run format` and passes the checks in `npm run check`
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/serialPortControl.ts [hunk 1]
```diff
@@ -18,21 +18,16 @@ interface ISerialPortDetail {
...
-                (error: any, ports: ISerialPortDetail[]) => {
-                    if (error) {
-                        reject(error);
-                    } else {
-                        resolve(ports);
-                    }
-                }
+            SerialPortControl.serialport.list().then(
+                ports => resolve(ports),
+                err => reject(err)
```
Comment (nasadigital) [inline src/serialPortControl.ts:30]: When you open the serial monitor without selecting serial port, the following pop up is shown:
Annotation 2020-03-19 125636
Do you think you could change the buttons from "Yes" and "No" to "Select port" and "Cancel" or something similar?

**Review by nasadigital — APPROVED** [review]
Awesome job, really streamlined this part!
LGTM ✌️

File: src/serialPortControl.ts (hunk 1 above)
Comment (vandyliu) [reply src/serialPortControl.ts:30]: I can definitely do that, nice catch!

File: package.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
     "instrumentationKey": "__AIKEY__",
     "icon": "assets/icon.png",
     "engines": {
-        "vscode": "^1.34.0"
+        "vscode": "^1.43.0"
```
Comment (xnkevinnguyen) [inline package.json:10]: Would it be worthwhile to add this into the readme ( if it's not already added)

Comment (vandyliu) [reply package.json:10]: sure, can do

Comment (vandyliu) [reply package.json:10]: Added it as a badge

**Review by andreamah — APPROVED** [review]
Tested and works :) Great to have the serial monitor back!

**Pull request merged**